alfalfa_web/server/api.js

```diff
@@ -104,306 +92,280 @@ class AlfalfaAPI {
-          MessageBody: JSON.stringify(messageBody),
-          QueueUrl: process.env.JOB_QUEUE_URL,
-          MessageGroupId: "Alfalfa"
-        })
-      );
-    } catch (e) {
-      console.error(e);
-      return Promise.reject();
-    }
+    const messageBody = {
+      job: `alfalfa_worker.jobs.${sim_type === "MODELICA" ? "modelica" : "openstudio"}.StepRun`,
+      params: {
+        run_id: run.ref_id,
+        start_datetime: startDatetime,
+        end_datetime: endDatetime,
+        timescale: `${timescale || 5}`,
+        realtime: `${!!realtime}`,
+        external_clock: `${!!externalClock}`
+      }
+    };
+    await this.sqs.send(
+      new SendMessageCommand({
+        MessageBody: JSON.stringify(messageBody),
+        QueueUrl: process.env.JOB_QUEUE_URL,
+        MessageGroupId: "Alfalfa"
+      })
+    );
   };
 
-  advanceRun = async (siteRef) => {
-    try {
-      return await this.sendRunMessage(siteRef, "advance");
-    } catch (e) {
-      console.error(e);
-      return Promise.reject();
-    }
+  advanceRun = async (run) => {
+    return await this.sendRunMessage(run, "advance");
   };
 
-  stopRun = async (siteRef) => {
-    try {
-      return await this.sendRunMessage(siteRef, "stop");
-    } catch (e) {
-      console.error(e);
-      return Promise.reject();
-    }
+  stopRun = async (run) => {
+    return await this.sendRunMessage(run, "stop");
   };
 
-  sendRunMessage = (siteRef, method, data = null, timeout = 6000, pollingInterval = 100) => {
-    return new Promise(async (resolve) => {
+  sendRunMessage = (run, method, data = null, timeout = 6000, pollingInterval = 100) => {
+    return new Promise(async (resolve, reject) => {
       const message_id = uuidv1();
-      await this.pub.publish(siteRef, JSON.stringify({ message_id, method, data }));
+      await this.pub.publish(run.ref_id, JSON.stringify({ message_id, method, data }));
       const send_time = Date.now();
 
       let interval;
 
       const finalize = (success, message = "") => {
         clearInterval(interval);
-        resolve({ status: success, message });
+        if (success) {
+          resolve({ message });
+        } else {
+          reject({ message });
+        }
       };
 
       interval = setInterval(async () => {
         if (Date.now() - timeout > send_time) {
           finalize(false, "no simulation reply");
         }
-        const response = await getHashValue(this.redis, siteRef, message_id);
+        const response = await getHashValue(this.redis, run.ref_id, message_id);
         if (!response) return;
         finalize(JSON.parse(response).status === "ok", response);
       }, pollingInterval);
     });
   };
 
   listModels = async () => {
-    try {
-      return (await this.models.find().toArray()).map(({ ref_id: id, model_name: modelName, created, modified }) => ({
-        id,
-        modelName,
-        created,
-        modified
-      }));
-    } catch (e) {
-      console.error(e);
```

Comment:
This kind of error handling was removed in favor of top level error handling to make the api more communicative about what went wrong.